File: haskell/src/Day5.hs
module Day5
    ( part1
    , part2
    )
where

import           Lib
import           IntcodeComputer

prepareInput input = do
    registries <- map (read :: String -> Int) <$> readInputForDayAsCSV 5
    return $ (fromRegistries registries) { input = input }

part1 = last . output . runProgramm <$> prepareInput [1]

part2 = last . output . runProgramm <$> prepareInput [5]

File: js/day5.js
require("./Problem")({
  input() {
    return require("./Input")
      .day(5)
      .split(",")
      .map(Number);
  },
  solve(registries) {
    const last = xs => xs[xs.length - 1];
    const program = require("./IntcodeComputer").program(registries);
    const part1 = program.withInput([1]).run();
    const part2 = program.withInput([5]).run();
    return {
      part1: last(part1.output),
      part2: last(part2.output)
    };
  }
}).run();
